include/bl_storage_internal.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BL_STORAGE_INTERNAL_H_
+#define BL_STORAGE_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup bl_storage Bootloader storage (protected data).
+ * @{
+ */
+
+const struct monotonic_counter *get_counter_struct(uint16_t description);
+
+/** This library implements monotonic counters where each time the counter
+ *  is increased, a new slot is written.
+ *  This way, the counter can be updated without erase. This is, among other things,
+ *  necessary so the counter can be used in the OTP section of the UICR
+ *  (available on e.g. nRF91 and nRF53).
+ */
+struct monotonic_counter {
+	/* Counter description. What the counter is used for. See
+	 * BL_MONOTONIC_COUNTERS_DESC_x.
+	 */
+	uint32_t description;
+	/* Number of entries in 'counter_slots' list. */
+	uint32_t num_counter_slots;
```

Comment:
These have been changed from uint16_t to uin32_t for all devices, so all APIs should be changed as well IMO, as well as changing reading from halfword_read to halfword_write.

Reply:
will change these back to uint16_t